### PR TITLE
Chore/gradle remove redundant apply

### PR DESCRIPTION
**Pull request by bartekpacia**
I'm removing an unneded block of configuration from the `settings.gradle` template. It was introduced by me in #123511. At that time, I did not know that it's unnecessary, and did not test removing it – sorry about that.

I learned that it's unnecessary recently, when [I asked a question on StackOverflow](https://stackoverflow.com/questions/77073596/whats-the-difference-between-plugins-and-pluginmanagement-plugins-in). More context there.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by gmackall — APPROVED** [review]
LGTM, thanks for the cleanup. Will need a test exemption

File: packages/flutter_tools/gradle/build.gradle.kts [hunk 1]
```diff
@@ -18,6 +18,7 @@ gradlePlugin {
             id = "dev.flutter.flutter-gradle-plugin"
             implementationClass = "FlutterPlugin"
         }
+        // The "flutterAppPluginLoaderPlugin" name isn't used anywhere.
```
Comment (Hixie) [inline packages/flutter_tools/gradle/build.gradle.kts:21]: what does this comment mean in practice?
(isn't it used on the very next line, for example?)

**Comment by Hixie** [comment]
test-exempt code removal

File: packages/flutter_tools/gradle/build.gradle.kts (hunk 1 above)
Comment (bartekpacia) [reply packages/flutter_tools/gradle/build.gradle.kts:21]: It means that `flutterAppPluginLoaderPlugin` can be changed to something other (for whatever reason) and it won't affect anything.

Nothing in the codebase references `flutterAppPluginLoaderPlugin`.

**Review by reidbaker — APPROVED** [review]

**Pull request merged**